components/txn_types/src/write.rs

```diff
@@ -53,6 +53,10 @@ impl WriteType {
             WriteType::Rollback => FLAG_ROLLBACK,
         }
     }
+
+    pub fn valid(&self) -> bool {
```

Comment:
However I think `valid` is not the best name here, it's kind of too wide, but I don't have a better one.

Reply:
Could you document what the function is for? I'm not sure what it does so I can't suggest a better name

Reply:
> Could you document what the function is for? I'm not sure what it does so I can't suggest a better name

The definition of `valid` can be viewed [here](https://github.com/tikv/tikv/blob/a68cb230a818f34b3316874a523bea8a2e17e8a7/src/storage/mvcc/reader/scanner/mod.rs#L388), however I'm afraid `valid` is not a good name for that function too, maybe we should make all these `valid` into `has_old_value` or something alike.

Reply:
I see. I'm not sure about that function, but I think this one would be better called `has_old_value` or similar